pkg/controller/bundlec_test/resolve_binding_secret_references_test.go

```diff
@@ -131,32 +104,77 @@ func TestResolveBindingSecretReferences(t *testing.T) {
 						},
 					},
 					{
-						Name: resSi2,
+						Name: "map-needs-update",
 						References: []smith_v1.Reference{
-							{Resource: smith_v1.ResourceName(resSb1)},
+							sb1ref,
 						},
 						Spec: smith_v1.ResourceSpec{
-							Object: &sc_v1b1.ServiceInstance{
+							Object: &core_v1.ConfigMap{
 								TypeMeta: meta_v1.TypeMeta{
-									Kind:       "ServiceInstance",
-									APIVersion: sc_v1b1.SchemeGroupVersion.String(),
+									Kind:       "ConfigMap",
+									APIVersion: core_v1.SchemeGroupVersion.String(),
 								},
 								ObjectMeta: meta_v1.ObjectMeta{
-									Name: si2,
+									Name: mapNeedsAnUpdate,
 									Annotations: map[string]string{
-										"Secret": "{{" + resSb1 + ":bindsecret#Data.mysecret}}",
```

Comment:
This test should have been failing but it was not. Bundle was updated to add the missing finalizer (see around line 64) and the reference was not processed. In fact, the reference format is old here - test test should have been updated when the references refactoring was performed. I've expanded this test to assert the bundle conditions to make sure it does not regress in the future. Also checked if all other tests do have the finalizer specified (there is one that does not intentionally - it tests the the finalizer is added).